### PR TITLE
feat/ OgTracker API Integration for Reports

### PR DESCRIPTION
**Pull request by VAIBHAV-25**
## Release Notes

- **New Features**
	- Introduced a new API function to fetch OG tracker data for posts.
	- Enhanced the `UserReportInfo` component with improved report uniqueness handling and conditional rendering based on report origin.
	- Added an API endpoint for adding or updating progress reports related to OG trackers.
	- Implemented tooltips in the `UploadMultipleReports` component to provide context for reports from OG Tracker.

- **Improvements**
	- Updated the rendering logic for progress report links to ensure better usability and presentation.
	- Conditional rendering of summary editing buttons based on report origin to streamline user interactions.
	- Simplified the rendering logic in the `Post` component for better maintainability.
	- Altered the logic for displaying upload options based on user identity.
	- Added a new property to indicate if reports originate from the OG tracker.

- **Bug Fixes**
	- Improved error handling in API endpoints for better user feedback.

### CODE REVIEW

File: pages/api/v1/posts/on-chain-post.ts [hunk 1]
```diff
@@ -119,6 +136,14 @@ interface IGetOnChainPostParams {
 	includeSubsquareComments?: boolean;
 }
 
+interface IOGData {
+	fdate: string;
```
Comment (alphainfinitus) [inline pages/api/v1/posts/on-chain-post.ts:140]: fdate is not self-explanatory or is a possible typo

File: src/components/ProgressReport/ProgressReportInfo/ReportDetails.tsx [hunk 2]
```diff
@@ -136,31 +136,43 @@ const ReportDetails: FC<IReportDetails> = (props) => {
...
-							alt='pdf.icon'
-						/>
-					</div>
-					<div className='flex flex-col gap-y-0.5'>
+			<div>
+				{report?.progress_file &&
+					(() => {
+						try {
+							const fileUrl = new URL(report.progress_file);
+							const allowedDomains = ['polkassembly.io', 'subsquare.io'];
```
Comment (alphainfinitus) [inline src/components/ProgressReport/ProgressReportInfo/ReportDetails.tsx:144]: this check should be in the back-end, not FE, we're using unnecessary network bandwidth

File: src/components/ProgressReport/ProgressReportInfo/UserReportInfo.tsx [hunk 3]
```diff
@@ -44,14 +45,40 @@ const UserReportInfo: FC<IUserReportInfo> = (props) => {
 	const dispatch = useDispatch();
 	const { open_rating_modal, open_rating_success_modal, report_rating } = useProgressReportSelector();
 
+	const uniqueReports = useMemo(() => {
+		if (!postData?.progress_report) return [];
+		return Object.entries(postData.progress_report as IProgressReport).sort((a, b) => new Date(a[1].created_at).getTime() - new Date(b[1].created_at).getTime());
```
Comment (alphainfinitus) [inline src/components/ProgressReport/ProgressReportInfo/UserReportInfo.tsx:50]: reading first element directly.. please add null checks a[1]

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

File: pages/api/v1/posts/on-chain-post.ts (hunk 1 above)
Comment (VAIBHAV-25) [reply pages/api/v1/posts/on-chain-post.ts:140]: sir, we are getting this field from ogTracker api and we are not setting this variable or creating this in our platform

**Review by alphainfinitus — APPROVED** [review]

**Pull request merged**